R/setup_docs.R

```diff
@@ -0,0 +1,104 @@
+#' Initialize documentation website settings
+#'
+#' @description
+#' Creates a subdirectory called `altdoc/` in the package root directory to store the settings files used to by one of the documentation generator utilities (docsify, docute, or mkdocs). The files in this folder are never altered automatically by `altdoc` unless the user explicitly calls `overwrite=TRUE`. They can thus be edited manually to customize the sidebar and website.
+#'
+#' @param tool String. "docsify", "docute", or "mkdocs".
+#' @param path Path to the package root directory.
+#' @param overwrite Logical. If TRUE, overwrite existing files. Warning: This will completely delete the settings files in the `altdocs` directory, including any customizations you may have made.
+#'
+#' @export
+#'
+#' @return No value returned.
+#' 
+#' @examples
+#' if (interactive()) {
+#' 
+#'   # Create docute documentation
+#'   setup_docs(tool = "docute")
+#'
+#'   # Create docsify documentation
+#'   setup_docs(tool = "docsify")
+#'
+#'   # Create mkdocs documentation
+#'   setup_docs(tool = "mkdocs")
+#' 
+#' }
+setup_docs <- function(tool, path = ".", overwrite = FALSE) {
+
+  safe.copy <- function(src, tar, overwrite) {
+    if (fs::file_exists(tar) && !isTRUE(overwrite)) {
+      cli::cli_abort("{tar} already exists. Delete it or set `overwrite=TRUE`.", call = NULL)
+    } else {
+      file.copy(src, tar, overwrite = TRUE)
+    }
+  }
+
+  # input sanity checks
+  if (missing(tool) ||
+      !is.character(tool) ||
+      length(tool) != 1 ||
+      !tool %in% c("docute", "docsify", "mkdocs")) {
+    cli::cli_abort(
+      'The `tool` argument must be "docsify", "docute", or "mkdocs".')
+  }
```

Comment:
doesn't `match.arg()` take care of all of this?

Reply:
Yes, but this error is more consistent with the "look" of the other messages.